supervisor/core.py

```diff
@@ -241,8 +242,16 @@ async def start(self):
                 self.sys_homeassistant.boot
                 and not await self.sys_homeassistant.core.is_running()
             ):
-                with suppress(HomeAssistantError):
+                try:
                     await self.sys_homeassistant.core.start()
+                except HomeAssistantCrashError as err:
+                    _LOGGER.warning("Can't start Home Assistant Core - rebuild")
+                    self.sys_capture_exception(err)
+
+                    with suppress(HomeAssistantError):
+                        await self.sys_homeassistant.core.rebuild()
```

Comment:
Should we check for space issues before throwing out the existing?

Reply:
We don't raise the Crash error if the diskspace is an issue to build the container

Reply:
I somehow thought that the rebuild pulled a new image.
But it only does a restart